lib/scarpe/logger.rb

```diff
@@ -83,6 +83,23 @@ def initialize_logger(log_config)
       end
     end
   end
+
+  class LoggedWrapper
+    include ::Scarpe::Log
+    def initialize(instance, component = instance)
+      log_init(component)
+
+      @instance = instance
+    end
+
+    def method_missing(method, ...)
+      self.singleton_class.define_method(method) do |*args, **kwargs, &block|
```

Comment:
This is amazing! So basically we grab the call, log that information out, and then send the call back on to our displayservice friend anyway. Amazing!

This repo has so many nuggest in it haha

Reply:
If we keep at it, there will be more. It's not a coincidence that nontrivial apps and frameworks tend to grow lots of little specific features. The default off-the-rack ones are fine, but often not perfect. And Ruby in particular feels the lack of Log4J or rough equivalent. A few libraries try to do that (including Logging, the one I started from here) but I don't feel like any of them *quite* get it.

I *want* there to be something like Logging (per-module logging) plus Cabin (structured and nested logging: https://github.com/jordansissel/ruby-cabin) plus a standard config file format. Then you could have little standard facilities like this on top of it. But there isn't.